### PR TITLE
fix: none container logger & port validation & moved in use warnings

### PR DESCRIPTION
**Pull request by robot9706**
- Fixed an issue with dagent trying to find a 'none' logger driver
- Fixed an issue in dagent crashing if it failed to create a container
- Added validation to ports and port ranges overlapping
- Moved registry and storage in use warnings under the card title
- Fixed an issue with config JSON editors not sending patches due to an error in `mergeKeyValuesWithJson`

### CODE REVIEW

File: web/crux/src/app/deploy/deploy.mapper.ts [hunk 1]
```diff
@@ -303,13 +303,14 @@ export default class DeployMapper {
   dagentConfigToAgentProto(config: MergedContainerConfigData): DagentContainerConfig {
     return {
       networks: this.mapUniqueKeyToStringArray(config.networks),
```
Comment (polaroi8d) [inline web/crux/src/app/deploy/deploy.mapper.ts:305]: Don't we have a mapper test for this?

Comment (robot9706) [reply web/crux/src/app/deploy/deploy.mapper.ts:305]: I don't think so.

File: web/crux/src/app/deploy/deploy.mapper.spec.ts [hunk 2]
```diff
@@ -821,4 +822,43 @@ describe('DeployMapper', () => {
       expect(config).toMatchObject(expected)
     })
   })
+
+  describe('dagentConfigToAgentProto logConfig', () => {
```
Comment (polaroi8d) [inline web/crux/src/app/deploy/deploy.mapper.spec.ts:826]: Nice 🥇

**Review by polaroi8d — APPROVED** [review]
LGTM (informal)

**Comment by ngyradi** [comment]
Ports overlap empty port ranges, after deleting port ranges it still says overlapping unless the reset button has been used.

**Comment by ngyradi** [comment]
works, tested locally

File: web/crux-ui/src/components/registries/edit-registry-card.tsx [hunk 3]
```diff
@@ -147,26 +147,28 @@ const EditRegistryCard = (props: EditRegistryCardProps) => {
         {editing ? t('common:editName', { name: registry.name }) : t('new')}
       </DyoHeading>
 
-      <DyoLabel className="text-light">{t('tips.common')}</DyoLabel>
+      <DyoLabel className="text-light block">{t('tips.common')}</DyoLabel>
+
+      {formik.values.inUse && (
+        <DyoLabel className="mt-2 block" textColor="text-sm text-warning-orange">
```
Comment (m8vago) [inline web/crux-ui/src/components/registries/edit-registry-card.tsx:153]: We have `DyoMessage` for this purposes.

File: web/crux-ui/src/components/storages/edit-storage-card.tsx [hunk 4]
```diff
@@ -89,6 +89,11 @@ const EditStorageCard = (props: EditStorageCardProps) => {
       <DyoHeading element="h4" className="text-lg text-bright">
         {editing ? t('common:editName', { name: storage.name }) : t('new')}
       </DyoHeading>
+      {formik.values.inUse && (
+        <DyoLabel className="mt-2" textColor="text-sm text-warning-orange">
```
Comment (m8vago) [inline web/crux-ui/src/components/storages/edit-storage-card.tsx:93]: same as above

File: web/crux-ui/src/validations/container.ts [hunk 5]
```diff
@@ -186,16 +188,41 @@ const storageRule = yup
...
-      external: portNumberOptionalRule,
-    }),
+const createOverlapTest = (
+  schema: yup.NumberSchema<number, AnyObject, number>,
+  portRanges: ContainerConfigPortRange[],
+  field: Exclude<keyof ContainerConfigPortRange, 'id'>,
+) =>
+  // eslint-disable-next-line no-template-curly-in-string
+  schema.test('port-range-overlap', '${path} overlaps port ranges', (value, _) =>
+    portRanges.length > 0 ? !portRanges.some(it => value >= it[field].from && value <= it[field].to) : true,
```
Comment (m8vago) [inline web/crux-ui/src/validations/container.ts:198]: Please make a body for the lambda and declare some const to make this more readable.

**Review by m8vago — CHANGES_REQUESTED** [review]

**Review by m8vago — APPROVED** [review]

**Pull request merged**